src/node-zk.cpp

```diff
@@ -296,7 +299,18 @@ class ZooKeeper: public Nan::ObjectWrap {
 
         if (rc) {
             LOG_ERROR("yield:zookeeper_interest returned error: %d - %s\n", rc, zerror(rc));
+            if(connection_state == 1 || connection_state == -1) {
+                connection_state = 0;
+                LOG_INFO("yield: disconnected state has been set: %d\n", connection_state);
+                DoEmitClose(Nan::New(on_disconnected), rc);
+            }
```

Comment:
A question: the `rc` response can be of several failure codes (in `include/zookeeper.h` of the C client this add-on is built upon). One of them is -1 (system error). How do we know this is a "disconnect"?

Reply:
Since the zookeeper_interest function is the main way this client communicates with the server over the network, if it’s unable to operate—no matter the reason—it indicates that the client is unable to maintain communication with the server.

Reply:
I was referring to the value of `rc` that can be any value other than `0`. This can mean several different things, and not always a disconnect. Have you looked into the possible values and what they mean?

Reply:
> I was referring to the value of `rc` that can be any value other than `0`. This can mean several different things, and not always a disconnect. Have you looked into the possible values and what they mean?

Thank you for the reminder.

I took another look at the function zookeeper_interest in ZooKeeper where rc is returned, and I realized that I overlooked an authentication-related code, which I believe is quite important. 

After careful consideration, I think `connect_failed` might be more appropriate. I will provide the details in a separate message. 

Thanks again for your attention and support.